native/core/src/events/attachments.rs

```diff
@@ -73,19 +92,31 @@ impl FallbackAttachmentContent {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct LinkAttachmentContent {
+    /// A short name for the given link
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub name: Option<String>,
+
+    /// The actual Link / URL / URI
+    pub link: String,
+}
+
 #[derive(Clone, Debug, Deserialize, Serialize)]
 #[serde(tag = "type")]
 pub enum AttachmentContent {
-    /// An image message.
+    /// An image attachment.
     Image(ImageMessageEventContent),
-    /// A video message.
+    /// A video attachment.
     Video(VideoMessageEventContent),
-    /// An audio message.
+    /// An audio attachment.
     Audio(AudioMessageEventContent),
-    /// A file message
+    /// A file attachment
     File(FileMessageEventContent),
-    /// A location message.
+    /// A location attachment.
     Location(LocationMessageEventContent),
+    /// A location attachment.
```

Comment:
You might want to use the following comment.
```suggestion
    /// A link attachment.
```